src/Type/Regex/RegexAstWalkResult.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Regex;
```

Comment:
```suggestion
namespace PHPStan\Type\Regex\AST;
```

the Hoa AST should be parsed into PHPStan AST with clean object design

walking such AST will be much easier and cleaner then